cpp/include/Ice/FactoryTable.h

```diff
@@ -88,27 +88,27 @@ namespace IceInternal
     template<class E> class DefaultUserExceptionFactoryInit
     {
     public:
-        DefaultUserExceptionFactoryInit(std::string_view tId) : typeId(tId)
+        DefaultUserExceptionFactoryInit(const char* tId) noexcept : typeId(tId)
         {
             factoryTable->addExceptionFactory(typeId, defaultUserExceptionFactory<E>);
         }
 
         ~DefaultUserExceptionFactoryInit() { factoryTable->removeExceptionFactory(typeId); }
 
-        const std::string_view typeId;
+        const char* typeId;
     };
 
     template<class O> class DefaultValueFactoryInit
     {
     public:
-        DefaultValueFactoryInit(std::string_view tId) : typeId(tId)
+        DefaultValueFactoryInit(const char* tId) noexcept : typeId(tId)
```

Comment:
The `std::string_view(const char*)` constructor is not noexcept - and we always construct these objects with string literals.